File: backend/src/Caro.Core.Domain/Entities/GameMode.cs
namespace Caro.Core.Domain.Entities;

/// <summary>
/// Game mode types for Caro matches.
/// </summary>
public enum GameMode
{
    PvP = 0,
    PvAI = 1,
    AivAI = 2
}

public static class GameModeExtensions
{
    public static string ToLowerString(this GameMode mode) => mode switch
    {
        GameMode.PvP => "pvp",
        GameMode.PvAI => "pvai",
        GameMode.AivAI => "aivai",
        _ => "pvp"
    };
}

File: backend/internal/domain/gamemode.go
package domain

type GameMode int

const (
	GameModePvP GameMode = iota
	GameModePvAI
	GameModeAivAI
)

func (m GameMode) String() string {
	switch m {
	case GameModePvP:
		return "pvp"
	case GameModePvAI:
		return "pvai"
	case GameModeAivAI:
		return "aivai"
	default:
		return "pvp"
	}
}

func ParseGameMode(s string) GameMode {
	switch s {
	case "pvai":
		return GameModePvAI
	case "aivai":
		return GameModeAivAI
	default:
		return GameModePvP
	}
}
